utility/accessdata.py

```diff
@@ -0,0 +1,190 @@
+                                                        body.find('common_local_data_directory').text)
+        self.user_local_data_directory = os.path.join(os.path.dirname(__file__), '..',
+                                                      body.find('user_local_data_directory').text)
+        self.server_address = body.find('server_address').text
+        self.server_user = body.find('user_name').text
+        self.server_private_access = body.find('server_private_data').text
+        self.server_public_access = body.find('server_public_data').text
+        self.server_public_address = body.find('server_public_address').text
+        self.contact_address = body.find('contact_address').text
+        self.private_key_path = body.find('private_key').text
+
+    def _set_private_connection(self):
+        self._set_private_key()
+        if self.private_key is not None:
+            self.client = paramiko.SSHClient()
+            self.client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        else:
+            self.client = None
+
+    def _set_private_key(self):
+        key_path = os.path.join(os.path.dirname(__file__), '..', self.private_key_path)
+        if os.path.isfile(key_path):
+            try:
+                self.private_key = paramiko.RSAKey.from_private_key_file(key_path)
+            except paramiko.SSHException:
+                self.private_key = None
+        else:
+            self.private_key = None
+
+    def _path_setup(self):
+        self._local_path_setup()
+        self._server_path_setup()
+
+    def _local_path_setup(self, local_path=None):
+        if local_path is None:
+            local_path = self.data_path_name
+        elif not isinstance(local_path, str):
+            raise TypeError('File local path input is expected to be of str type.')
+        self.common_local_data_path = os.path.join(self.common_local_data_directory, local_path)
+        self.user_local_data_path = os.path.join(self.user_local_data_directory, local_path)
+        self.user_local_dummy_path = os.path.join(self.user_local_data_directory,
+                                                  self.dummy_directory, local_path)
+
+    def _server_path_setup(self, server_path=None):
+        if server_path is None:
+            server_path = self.data_path_name
+        elif not isinstance(server_path, str):
+            raise TypeError('File local path input is expected to be of str type.')
+        self.server_public_path = self._set_public_server_path(server_path)
+        self.server_private_path = self._set_private_server_path(server_path)
+        self.server_to_public_path = self._set_public_server_access_path(server_path)
+
+    def _set_private_server_path(self, path):
+        return self.server_private_access + '/' + path
+
+    def _set_public_server_path(self, path):
+        return self.server_public_address + '/' + path
+
+    def _set_public_server_access_path(self, path):
```

Comment:
rename all:
*public_server_access_path* --> *public_server_write_access_path*

Reply:
Refactored the root handling variables: 5e76fd1

Reply:
Refactored generator routine: d6cd6aa

Reply:
Refactored property: e53d868

Reply:
All refactoring done.